core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -448,6 +448,7 @@ class BrokerServer(
       try {
         metadataListener.startPublishing(metadataPublisher).get()
       } catch {
+        case ee: ExecutionException => throw ee
         case t: Throwable => throw new RuntimeException("Received a fatal error while " +
           "waiting for the broker to catch up with the current cluster metadata.", t)
```

Comment:
I think the reason we'd like to wrap the original exception is to add some custom messages. I'm afraid after this change, it'll lose it and just throw ExecutionException directly.  

I'm thinking, maybe in L485, we just do:
```
throw if (e.isInstanceOf[RuntimeException]) e.getCause.getCause else e
```
if we confirmed all the exceptions are wrapped with `RuntimeException`, Or maybe just `throw e` and don't care about `getCause`. WDYT?

Reply:
> Or maybe just throw e and don't care about getCause

This implemented right now in trunk :)

> if we confirmed all the exceptions are wrapped with RuntimeException

`CompletableFuture.get` can throw different type of exceptions. From javadoc - `CancellationException`, `InterruptedException`.

> I'm afraid after this change, it'll lose it and just throw ExecutionException directly.

I thought that is a goal of the [KAFKA-13908](https://issues.apache.org/jira/browse/KAFKA-13908).
To throw original exception not a wrapper.

> throw if (e.isInstanceOf[RuntimeException]) e.getCause.getCause else e

RuntimeException can be throwed from the other lines of try block, so this logic can cause NPE.

Reply:
> I think the reason we'd like to wrap the original exception is to add some custom messages. I'm afraid after this change, it'll lose it and just throw ExecutionException directly.

Changes preserve behaviour before #11969 
If `lifecycleManager.setReadyToUnfence().get()` or `metadataListener.startPublishing(metadataPublisher).get()` throw `ExecutionException` it will be passed bo global catch block and unwrapped. Otherwise, in case of error in those lines wrapped `RuntimeException` will be the case.


@showuon Can you, please, clarify. Am I understand the goal of the ticket correctly? We want to preserve "old" behaviour?

Reply:
@nizhikov , thanks for the response. Here's my understanding:
Before this PR, when it failed in 
L473: `lifecycleManager.setReadyToUnfence().get()`, we'll get 
```
Fatal error during broker startup. Prepare to shutdown.
java.lang.RuntimeException: Received a fatal error while waiting for the broker to be unfenced.
 at ....
 at ....

...
Exiting Kafka due to fatal exception during startup.
java.lang.RuntimeException: Received a fatal error while waiting for the broker to be unfenced.
 at ....
 at ....
```

But after your PR, it will not show the message: `Received a fatal error while waiting for the broker to be unfenced` in both top/bottom logs, which is unexpected.

My expectation is after this PR, we can display the cause of the `RuntimeException` in the below log output, ex: 
```
```
Exiting Kafka due to fatal exception during startup.
java.lang.IOException: Not enough disk space
 at ....
 at ....
```
If that will make the logic too complicated, maybe we'll just leave it as is. Thoughts?